src/pages/SignIn/components/UserSignIn.jsx

```diff
@@ -1,42 +1,79 @@
 import { signInWithEmailAndPassword } from 'firebase/auth';
-import React, { useRef } from 'react';
+import React, { useRef, useState } from 'react';
 import { auth } from '../../../utils/firebase';
 import { Input, Button } from '@nextui-org/react';
+import { EyeFilled, EyeInvisibleFilled } from '@ant-design/icons';
 
 const UserSignIn = () => {
   const email = useRef();
   const password = useRef();
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isInvalid, setIsInvalid] = useState(false);
+  const [isVisible, setIsVisible] = React.useState(false);
+
+  const toggleVisibility = () => setIsVisible(!isVisible);
```

Comment:
i believe `setIsVisible((prev) => !prev)` is more safer and better way

Reply:
done